diff --git a/wmlib/src/WindowsMediaLib.csproj b/wmlib/src/WindowsMediaLib.csproj
index 24c8569..4aaf700 100644
--- a/wmlib/src/WindowsMediaLib.csproj
+++ b/wmlib/src/WindowsMediaLib.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net47;net20</TargetFrameworks>
+    <TargetFrameworks>netstandard2.1</TargetFrameworks>
     <Version>1.1.0.0</Version>
     <AssemblyName>WindowsMediaLib</AssemblyName>
     <RootNamespace>WindowsMediaLib</RootNamespace>
@@ -10,7 +10,9 @@
     <Copyright>Copyright (c) 2008-2018 snarfle</Copyright>
     <Authors>snarfle</Authors>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <OutputPath>$(SolutionDir)\..\bin\$(Configuration)\plugins</OutputPath>
+    <OutputPath>$(CueToolsBinRoot)plugins\</OutputPath>
+    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
+    <IntermediateOutputPath>$(BaseIntermediateOutputPath)$(Configuration)\$(TargetFramework)\</IntermediateOutputPath>
     <RepositoryUrl>https://svn.code.sf.net/p/windowsmedianet/code/</RepositoryUrl>
     <RepositoryType>svn</RepositoryType>
     <Company />
