diff -ur 0_MAC_904_SDK_orig/Source/MACLib/APECompress.cpp 1_MAC_904_SDK_updated/Source/MACLib/APECompress.cpp
--- 0_MAC_904_SDK_orig/Source/MACLib/APECompress.cpp	2022-12-09 18:31:04.000000000 +0100
+++ 1_MAC_904_SDK_updated/Source/MACLib/APECompress.cpp	2022-12-28 12:39:50.000000000 +0100
@@ -32,6 +32,7 @@
     }
 }
 
+#ifndef EXCLUDE_CIO
 int CAPECompress::Start(const wchar_t * pOutputFilename, const WAVEFORMATEX * pwfeInput, int64 nMaxAudioBytes, int nCompressionLevel, const void * pHeaderData, int64 nHeaderBytes, int nFlags)
 {
     if (m_pioOutput)
@@ -57,6 +58,7 @@
 
     return nStartResult;
 }
+#endif
 
 int CAPECompress::StartEx(CIO * pioOutput, const WAVEFORMATEX * pwfeInput, int64 nMaxAudioBytes, int nCompressionLevel, const void * pHeaderData, int64 nHeaderBytes)
 {
diff -ur 0_MAC_904_SDK_orig/Source/MACLib/APECompress.h 1_MAC_904_SDK_updated/Source/MACLib/APECompress.h
--- 0_MAC_904_SDK_orig/Source/MACLib/APECompress.h	2022-12-08 21:25:12.000000000 +0100
+++ 1_MAC_904_SDK_updated/Source/MACLib/APECompress.h	2022-12-28 13:26:58.000000000 +0100
@@ -20,7 +20,9 @@
     ~CAPECompress();
 
     // start encoding
+#ifndef EXCLUDE_CIO
     int Start(const wchar_t * pOutputFilename, const WAVEFORMATEX * pwfeInput, int64 nMaxAudioBytes, int nCompressionLevel = MAC_COMPRESSION_LEVEL_NORMAL, const void * pHeaderData = APE_NULL, int64 nHeaderBytes = CREATE_WAV_HEADER_ON_DECOMPRESSION, int nFlags = 0) APE_OVERRIDE;
+#endif
     int StartEx(CIO * pioOutput, const WAVEFORMATEX * pwfeInput, int64 nMaxAudioBytes, int nCompressionLevel = MAC_COMPRESSION_LEVEL_NORMAL, const void * pHeaderData = APE_NULL, int64 nHeaderBytes = CREATE_WAV_HEADER_ON_DECOMPRESSION) APE_OVERRIDE;
 
     // add data / compress data
diff -ur 0_MAC_904_SDK_orig/Source/MACLib/APEInfo.cpp 1_MAC_904_SDK_updated/Source/MACLib/APEInfo.cpp
--- 0_MAC_904_SDK_orig/Source/MACLib/APEInfo.cpp	2022-12-09 19:02:32.000000000 +0100
+++ 1_MAC_904_SDK_updated/Source/MACLib/APEInfo.cpp	2022-12-28 12:39:50.000000000 +0100
@@ -46,6 +46,7 @@
 {
 }
 
+#ifndef EXCLUDE_CIO
 /**************************************************************************************************
 Construction
 **************************************************************************************************/
@@ -115,6 +116,7 @@
     // update
     CheckHeaderInformation();
 }
+#endif
 
 CAPEInfo::CAPEInfo(int * pErrorCode, CIO * pIO, CAPETag * pTag)
 {
diff -ur 0_MAC_904_SDK_orig/Source/MACLib/APETag.cpp 1_MAC_904_SDK_updated/Source/MACLib/APETag.cpp
--- 0_MAC_904_SDK_orig/Source/MACLib/APETag.cpp	2022-12-09 18:57:02.000000000 +0100
+++ 1_MAC_904_SDK_updated/Source/MACLib/APETag.cpp	2022-12-28 12:39:50.000000000 +0100
@@ -125,6 +125,7 @@
     L"Crossover", L"Contemporary C", L"Christian Rock", L"Merengue", L"Salsa", L"Thrash Metal", L"Anime", L"JPop", L"SynthPop"
 };
 
+#ifndef EXCLUDE_CIO
 CAPETag::CAPETag(const str_utfn * pFilename, bool bAnalyze)
 {
     m_spIO.Assign(CreateCIO());
@@ -144,6 +145,7 @@
     if (bAnalyze)
         Analyze();
 }
+#endif
 
 CAPETag::CAPETag(CIO * pIO, bool bAnalyze, bool bCheckForID3v1)
 {
diff -ur 0_MAC_904_SDK_orig/Source/MACLib/MACLib.cpp 1_MAC_904_SDK_updated/Source/MACLib/MACLib.cpp
--- 0_MAC_904_SDK_orig/Source/MACLib/MACLib.cpp	2022-12-09 19:11:04.000000000 +0100
+++ 1_MAC_904_SDK_updated/Source/MACLib/MACLib.cpp	2023-01-03 10:16:08.000000000 +0100
@@ -75,6 +75,7 @@
     return spAPEDecompress.GetPtr();
 }
 
+#ifndef EXCLUDE_CIO
 IAPEDecompress * __stdcall CreateIAPEDecompress(const str_utfn * pFilename, int * pErrorCode, bool bReadOnly, bool bAnalyzeTagNow, bool bReadWholeFile)
 {
     // error check the parameters
@@ -138,6 +139,7 @@
     // return
     return pAPEDecompress;
 }
+#endif
 
 IAPEDecompress * __stdcall CreateIAPEDecompressEx(CIO * pIO, int * pErrorCode)
 {
@@ -255,6 +257,7 @@
     catch(...) { return ERROR_UNDEFINED; }
 }
 
+#ifndef EXCLUDE_CIO
 int __stdcall GetAPEFileType(const APE::str_utfn * pInputFilename, APE::str_ansi cFileType[8])
 {
     memset(&cFileType[0], 0, sizeof(APE::str_ansi) * 8);
@@ -889,3 +892,4 @@
     // return
     return nFunctionRetVal;
 }
+#endif
diff -ur 0_MAC_904_SDK_orig/Source/MACLib/MACLib.h 1_MAC_904_SDK_updated/Source/MACLib/MACLib.h
--- 0_MAC_904_SDK_orig/Source/MACLib/MACLib.h	2022-12-09 19:11:04.000000000 +0100
+++ 1_MAC_904_SDK_updated/Source/MACLib/MACLib.h	2022-12-28 13:28:24.000000000 +0100
@@ -375,9 +375,11 @@
     //        on decompression)
     //////////////////////////////////////////////////////////////////////////////////////////////
 
+#ifndef EXCLUDE_CIO
     virtual int Start(const str_utfn * pOutputFilename, const WAVEFORMATEX * pwfeInput,
         int64 nMaxAudioBytes = MAX_AUDIO_BYTES_UNKNOWN, int nCompressionLevel = MAC_COMPRESSION_LEVEL_NORMAL,
         const void * pHeaderData = APE_NULL, int64 nHeaderBytes = CREATE_WAV_HEADER_ON_DECOMPRESSION, int nFlags = 0) = 0;
+#endif
 
     virtual int StartEx(CIO * pioOutput, const WAVEFORMATEX * pwfeInput,
         int64 nMaxAudioBytes = MAX_AUDIO_BYTES_UNKNOWN, int nCompressionLevel = MAC_COMPRESSION_LEVEL_NORMAL,
diff -ur 0_MAC_904_SDK_orig/Source/Projects/VS2022/MACLib/MACLib.vcxproj 1_MAC_904_SDK_updated/Source/Projects/VS2022/MACLib/MACLib.vcxproj
--- 0_MAC_904_SDK_orig/Source/Projects/VS2022/MACLib/MACLib.vcxproj	2022-12-09 21:48:20.000000000 +0100
+++ 1_MAC_904_SDK_updated/Source/Projects/VS2022/MACLib/MACLib.vcxproj	2023-01-26 21:48:42.000000000 +0100
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project DefaultTargets="Build" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project DefaultTargets="Build" ToolsVersion="Current" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemGroup Label="ProjectConfigurations">
     <ProjectConfiguration Include="Debug|Win32">
       <Configuration>Debug</Configuration>
@@ -21,33 +21,31 @@
   <PropertyGroup Label="Globals">
     <ProjectGuid>{21BF980F-C022-4DCC-9250-7C73528E422B}</ProjectGuid>
     <RootNamespace>MACLib</RootNamespace>
-    <WindowsTargetPlatformVersion>10.0.22621.0</WindowsTargetPlatformVersion>
+    <!-- Latest Target Version property -->
+    <LatestTargetPlatformVersion>$([Microsoft.Build.Utilities.ToolLocationHelper]::GetLatestSDKTargetPlatformVersion('Windows', '10.0'))</LatestTargetPlatformVersion>
+    <WindowsTargetPlatformVersion Condition="'$(WindowsTargetPlatformVersion)' == ''">$(LatestTargetPlatformVersion)</WindowsTargetPlatformVersion>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
-    <PlatformToolset>v143</PlatformToolset>
-    <UseOfMfc>false</UseOfMfc>
+    <PlatformToolset>$(DefaultPlatformToolset)</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
     <WholeProgramOptimization>true</WholeProgramOptimization>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
-    <PlatformToolset>v143</PlatformToolset>
-    <UseOfMfc>false</UseOfMfc>
+    <PlatformToolset>$(DefaultPlatformToolset)</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
-    <PlatformToolset>v143</PlatformToolset>
-    <UseOfMfc>false</UseOfMfc>
+    <PlatformToolset>$(DefaultPlatformToolset)</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
     <WholeProgramOptimization>true</WholeProgramOptimization>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
     <ConfigurationType>StaticLibrary</ConfigurationType>
-    <PlatformToolset>v143</PlatformToolset>
-    <UseOfMfc>false</UseOfMfc>
+    <PlatformToolset>$(DefaultPlatformToolset)</PlatformToolset>
     <CharacterSet>Unicode</CharacterSet>
     <WholeProgramOptimization>false</WholeProgramOptimization>
   </PropertyGroup>
@@ -75,29 +73,29 @@
     <_ProjectFileVersion>11.0.50727.1</_ProjectFileVersion>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
-    <OutDir>$(Configuration)\</OutDir>
-    <IntDir>$(Configuration)\</IntDir>
+    <OutDir>$(SolutionDir)..\ThirdParty$(Configuration)\$(Platform)\</OutDir>
+    <IntDir>$(SolutionDir)..\obj\$(ProjectName)\$(Platform)\$(Configuration)\</IntDir>
     <EnableClangTidyCodeAnalysis>true</EnableClangTidyCodeAnalysis>
     <ClangTidyChecks>-clang-diagnostic-c++98-compat,-clang-diagnostic-language-extension-token</ClangTidyChecks>
     <CodeAnalysisRuleSet>NativeRecommendedRules.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
-    <OutDir>x64\$(Configuration)\</OutDir>
-    <IntDir>x64\$(Configuration)\</IntDir>
+    <OutDir>$(SolutionDir)..\ThirdParty$(Configuration)\$(Platform)\</OutDir>
+    <IntDir>$(SolutionDir)..\obj\$(ProjectName)\$(Platform)\$(Configuration)\</IntDir>
     <EnableClangTidyCodeAnalysis>true</EnableClangTidyCodeAnalysis>
     <ClangTidyChecks>-clang-diagnostic-c++98-compat,-clang-diagnostic-language-extension-token</ClangTidyChecks>
     <CodeAnalysisRuleSet>NativeRecommendedRules.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
-    <OutDir>$(Configuration)\</OutDir>
-    <IntDir>$(Configuration)\</IntDir>
+    <OutDir>$(SolutionDir)..\ThirdParty\$(Platform)\</OutDir>
+    <IntDir>$(SolutionDir)..\obj\$(ProjectName)\$(Platform)\$(Configuration)\</IntDir>
     <EnableClangTidyCodeAnalysis>true</EnableClangTidyCodeAnalysis>
     <ClangTidyChecks>-clang-diagnostic-c++98-compat,-clang-diagnostic-language-extension-token</ClangTidyChecks>
     <CodeAnalysisRuleSet>NativeRecommendedRules.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
-    <OutDir>x64\$(Configuration)\</OutDir>
-    <IntDir>x64\$(Configuration)\</IntDir>
+    <OutDir>$(SolutionDir)..\ThirdParty\$(Platform)\</OutDir>
+    <IntDir>$(SolutionDir)..\obj\$(ProjectName)\$(Platform)\$(Configuration)\</IntDir>
     <EnableClangTidyCodeAnalysis>true</EnableClangTidyCodeAnalysis>
     <ClangTidyChecks>-clang-diagnostic-c++98-compat,-clang-diagnostic-language-extension-token</ClangTidyChecks>
     <CodeAnalysisRuleSet>NativeRecommendedRules.ruleset</CodeAnalysisRuleSet>
@@ -106,14 +104,10 @@
     <ClCompile>
       <Optimization>Disabled</Optimization>
       <AdditionalIncludeDirectories>..\..\..\MACLib;..\..\..\Shared;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <PreprocessorDefinitions>PLATFORM_WINDOWS;WIN32;_DEBUG;_LIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>EXCLUDE_CIO;PLATFORM_WINDOWS;WIN32;_DEBUG;_LIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <BasicRuntimeChecks>EnableFastChecks</BasicRuntimeChecks>
-      <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
-      <TreatWChar_tAsBuiltInType>true</TreatWChar_tAsBuiltInType>
-      <PrecompiledHeaderOutputFile>$(IntDir)$(TargetName).pch</PrecompiledHeaderOutputFile>
-      <AssemblerListingLocation>$(IntDir)</AssemblerListingLocation>
-      <ObjectFileName>$(IntDir)</ObjectFileName>
-      <ProgramDataBaseFileName>$(OutDir)$(TargetName).pdb</ProgramDataBaseFileName>
+      <RuntimeLibrary>MultiThreadedDebugDLL</RuntimeLibrary>
+      <TreatWChar_tAsBuiltInType>false</TreatWChar_tAsBuiltInType>
       <WarningLevel>EnableAllWarnings</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
@@ -131,7 +125,6 @@
       <Command />
     </PreLinkEvent>
     <Lib>
-      <OutputFile>$(OutDir)$(TargetName)$(TargetExt)</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <LinkTimeCodeGeneration>true</LinkTimeCodeGeneration>
     </Lib>
@@ -143,14 +136,10 @@
     <ClCompile>
       <Optimization>Disabled</Optimization>
       <AdditionalIncludeDirectories>..\..\..\MACLib;..\..\..\Shared;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <PreprocessorDefinitions>PLATFORM_WINDOWS;WIN32;_DEBUG;_LIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>EXCLUDE_CIO;PLATFORM_WINDOWS;WIN32;_DEBUG;_LIB;PLATFORM_x64;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <BasicRuntimeChecks>EnableFastChecks</BasicRuntimeChecks>
-      <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
-      <TreatWChar_tAsBuiltInType>true</TreatWChar_tAsBuiltInType>
-      <PrecompiledHeaderOutputFile>$(IntDir)$(TargetName).pch</PrecompiledHeaderOutputFile>
-      <AssemblerListingLocation>$(IntDir)</AssemblerListingLocation>
-      <ObjectFileName>$(IntDir)</ObjectFileName>
-      <ProgramDataBaseFileName>$(OutDir)$(TargetName).pdb</ProgramDataBaseFileName>
+      <RuntimeLibrary>MultiThreadedDebugDLL</RuntimeLibrary>
+      <TreatWChar_tAsBuiltInType>false</TreatWChar_tAsBuiltInType>
       <WarningLevel>EnableAllWarnings</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
@@ -168,40 +157,28 @@
       <Command />
     </PreLinkEvent>
     <Lib>
-      <OutputFile>$(OutDir)$(TargetName)$(TargetExt)</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <LinkTimeCodeGeneration>true</LinkTimeCodeGeneration>
     </Lib>
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
     <ClCompile>
-      <Optimization>MaxSpeed</Optimization>
-      <InlineFunctionExpansion>AnySuitable</InlineFunctionExpansion>
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <FavorSizeOrSpeed>Speed</FavorSizeOrSpeed>
       <OmitFramePointers>true</OmitFramePointers>
-      <EnableFiberSafeOptimizations>true</EnableFiberSafeOptimizations>
       <WholeProgramOptimization>true</WholeProgramOptimization>
       <AdditionalIncludeDirectories>..\..\..\MACLib;..\..\..\Shared;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <PreprocessorDefinitions>PLATFORM_WINDOWS;WIN32;NDEBUG;_LIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <StringPooling>true</StringPooling>
-      <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
+      <PreprocessorDefinitions>EXCLUDE_CIO;_CRT_SECURE_NO_WARNINGS;PLATFORM_WINDOWS;WIN32;NDEBUG;_LIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>
       <BufferSecurityCheck>false</BufferSecurityCheck>
-      <FunctionLevelLinking>true</FunctionLevelLinking>
-      <DisableLanguageExtensions>false</DisableLanguageExtensions>
-      <TreatWChar_tAsBuiltInType>true</TreatWChar_tAsBuiltInType>
       <RuntimeTypeInfo>false</RuntimeTypeInfo>
       <PrecompiledHeader />
       <PrecompiledHeaderFile>all.h</PrecompiledHeaderFile>
-      <PrecompiledHeaderOutputFile>$(IntDir)$(TargetName).pch</PrecompiledHeaderOutputFile>
-      <AssemblerListingLocation>$(IntDir)</AssemblerListingLocation>
-      <ObjectFileName>$(IntDir)</ObjectFileName>
-      <ProgramDataBaseFileName>$(OutDir)$(TargetName).pdb</ProgramDataBaseFileName>
-      <BrowseInformation>false</BrowseInformation>
       <WarningLevel>EnableAllWarnings</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <CompileAs>Default</CompileAs>
       <EnableEnhancedInstructionSet>StreamingSIMDExtensions2</EnableEnhancedInstructionSet>
+      <FloatingPointModel>Fast</FloatingPointModel>
     </ClCompile>
     <ResourceCompile>
       <PreprocessorDefinitions>NDEBUG;%(PreprocessorDefinitions)</PreprocessorDefinitions>
@@ -211,7 +188,6 @@
       <Command />
     </PreLinkEvent>
     <Lib>
-      <OutputFile>$(OutDir)$(TargetName)$(TargetExt)</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <LinkTimeCodeGeneration>true</LinkTimeCodeGeneration>
     </Lib>
@@ -221,7 +197,6 @@
       <TargetEnvironment>X64</TargetEnvironment>
     </Midl>
     <ClCompile>
-      <Optimization>MaxSpeed</Optimization>
       <InlineFunctionExpansion>AnySuitable</InlineFunctionExpansion>
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <FavorSizeOrSpeed>Speed</FavorSizeOrSpeed>
@@ -229,24 +204,16 @@
       <EnableFiberSafeOptimizations>true</EnableFiberSafeOptimizations>
       <WholeProgramOptimization>true</WholeProgramOptimization>
       <AdditionalIncludeDirectories>..\..\..\MACLib;..\..\..\Shared;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <PreprocessorDefinitions>PLATFORM_WINDOWS;WIN32;NDEBUG;_LIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <StringPooling>true</StringPooling>
-      <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
+      <PreprocessorDefinitions>EXCLUDE_CIO;_CRT_SECURE_NO_WARNINGS;PLATFORM_WINDOWS;WIN32;NDEBUG;_LIB;PLATFORM_x64;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <RuntimeLibrary>MultiThreadedDLL</RuntimeLibrary>
       <BufferSecurityCheck>false</BufferSecurityCheck>
-      <FunctionLevelLinking>true</FunctionLevelLinking>
-      <DisableLanguageExtensions>false</DisableLanguageExtensions>
-      <TreatWChar_tAsBuiltInType>true</TreatWChar_tAsBuiltInType>
       <RuntimeTypeInfo>false</RuntimeTypeInfo>
       <PrecompiledHeader />
       <PrecompiledHeaderFile>all.h</PrecompiledHeaderFile>
-      <PrecompiledHeaderOutputFile>$(IntDir)$(TargetName).pch</PrecompiledHeaderOutputFile>
-      <AssemblerListingLocation>$(IntDir)</AssemblerListingLocation>
-      <ObjectFileName>$(IntDir)</ObjectFileName>
-      <ProgramDataBaseFileName>$(OutDir)$(TargetName).pdb</ProgramDataBaseFileName>
-      <BrowseInformation>false</BrowseInformation>
       <WarningLevel>EnableAllWarnings</WarningLevel>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <CompileAs>Default</CompileAs>
+      <FloatingPointModel>Fast</FloatingPointModel>
       <EnableEnhancedInstructionSet>NotSet</EnableEnhancedInstructionSet>
     </ClCompile>
     <ResourceCompile>
@@ -257,7 +224,6 @@
       <Command />
     </PreLinkEvent>
     <Lib>
-      <OutputFile>$(OutDir)$(TargetName)$(TargetExt)</OutputFile>
       <SuppressStartupBanner>true</SuppressStartupBanner>
       <LinkTimeCodeGeneration>true</LinkTimeCodeGeneration>
     </Lib>
@@ -282,19 +248,44 @@
     <ClCompile Include="..\..\..\MACLib\UnBitArrayBase.cpp" />
     <ClCompile Include="..\..\..\MACLib\APEHeader.cpp" />
     <ClCompile Include="..\..\..\MACLib\APEInfo.cpp" />
-    <ClCompile Include="..\..\..\MACLib\APELink.cpp" />
+    <ClCompile Include="..\..\..\MACLib\APELink.cpp">
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Release|x64'">true</ExcludedFromBuild>
+    </ClCompile>
     <ClCompile Include="..\..\..\MACLib\APETag.cpp" />
-    <ClCompile Include="..\..\..\MACLib\WAVInputSource.cpp" />
+    <ClCompile Include="..\..\..\MACLib\WAVInputSource.cpp">
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Release|x64'">true</ExcludedFromBuild>
+    </ClCompile>
     <ClCompile Include="..\..\..\Shared\CharacterHelper.cpp" />
     <ClCompile Include="..\..\..\Shared\CircleBuffer.cpp" />
     <ClCompile Include="..\..\..\Shared\CRC.cpp" />
     <ClCompile Include="..\..\..\Shared\GlobalFunctions.cpp" />
-    <ClCompile Include="..\..\..\MACLib\MACProgressHelper.cpp" />
+    <ClCompile Include="..\..\..\MACLib\MACProgressHelper.cpp">
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Release|x64'">true</ExcludedFromBuild>
+    </ClCompile>
     <ClCompile Include="..\..\..\MACLib\MD5.cpp" />
     <ClCompile Include="..\..\..\MACLib\Prepare.cpp" />
     <ClCompile Include="..\..\..\Shared\IO.cpp" />
-    <ClCompile Include="..\..\..\Shared\StdLibFileIO.cpp" />
-    <ClCompile Include="..\..\..\Shared\WinFileIO.cpp" />
+    <ClCompile Include="..\..\..\Shared\StdLibFileIO.cpp">
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Release|x64'">true</ExcludedFromBuild>
+    </ClCompile>
+    <ClCompile Include="..\..\..\Shared\WinFileIO.cpp">
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">true</ExcludedFromBuild>
+      <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='Release|x64'">true</ExcludedFromBuild>
+    </ClCompile>
     <ClCompile Include="..\..\..\MACLib\NewPredictor.cpp" />
     <ClCompile Include="..\..\..\MACLib\NNFilter.cpp" />
     <ClCompile Include="..\..\..\Shared\BufferIO.cpp" />
diff -ur 0_MAC_904_SDK_orig/Source/Shared/All.h 1_MAC_904_SDK_updated/Source/Shared/All.h
--- 0_MAC_904_SDK_orig/Source/Shared/All.h	2022-12-11 16:30:58.000000000 +0100
+++ 1_MAC_904_SDK_updated/Source/Shared/All.h	2022-12-28 13:32:00.000000000 +0100
@@ -68,7 +68,21 @@
     #ifdef _MSC_VER
         #pragma warning(push) // push and pop warnings because the windows includes suppresses some like 4514
     #endif
+#if defined(PLATFORM_WINDOWS_NO_HEADER)
+    #define NEAR
+    #define FAR
+    #define MAX_PATH    4096
+    #define BOOL int
+    #define TRUE 1
+    #define FALSE 0
+    #define HWND long
+    typedef void * HANDLE;
+    #include <IntSafe.h>
+    #include <string.h>
+    #define ZeroMemory(p,l) memset((p), 0, (l))
+#else
     #include <Windows.h>
+#endif
     #ifdef _MSC_VER
         #pragma warning(pop)
     #endif
@@ -209,9 +223,13 @@
 #if defined(PLATFORM_WINDOWS)
     #define IO_USE_WIN_FILE_IO
     #define DLLEXPORT                                   __declspec(dllexport)
+#if defined(PLATFORM_WINDOWS_NO_HEADER)
+    #define SLEEP(MILLISECONDS)
+#else
     #define SLEEP(MILLISECONDS)                         ::Sleep(MILLISECONDS)
     #define MESSAGEBOX(PARENT, TEXT, CAPTION, TYPE)     ::MessageBox(PARENT, TEXT, CAPTION, TYPE)
     #define PUMP_MESSAGE_LOOP                           { MSG Msg; while (PeekMessage(&Msg, APE_NULL, 0, 0, PM_REMOVE) != 0) { TranslateMessage(&Msg); DispatchMessage(&Msg); } }
+#endif
     #define ODS                                         OutputDebugString
     #define TICK_COUNT_TYPE                             unsigned long long
     #define TICK_COUNT_READ(VARIABLE)                   VARIABLE = GetTickCount64()
diff -ur 0_MAC_904_SDK_orig/Source/Shared/GlobalFunctions.cpp 1_MAC_904_SDK_updated/Source/Shared/GlobalFunctions.cpp
--- 0_MAC_904_SDK_orig/Source/Shared/GlobalFunctions.cpp	2022-12-09 21:49:54.000000000 +0100
+++ 1_MAC_904_SDK_updated/Source/Shared/GlobalFunctions.cpp	2022-12-28 13:35:08.000000000 +0100
@@ -38,6 +38,7 @@
     return nResult;
 }
 
+#ifndef EXCLUDE_CIO
 bool FileExists(const wchar_t * pFilename)
 {
     if (0 == wcscmp(pFilename, L"-")  ||  0 == wcscmp(pFilename, L"/dev/stdin"))
@@ -73,6 +74,7 @@
 
 #endif
 }
+#endif
 
 void * AllocateAligned(intn nBytes, intn nAlignment)
 {
